src/main/java/fr/insee/pearljam/batch/utils/XmlUtils.java

```
@@ -216,6 +219,64 @@ public static void updateSampleFileErrorList(StreamResult sr, String fileName) {
+
+				if (variablesElement != null) { // will never be null
+					Courrier courrier = courriersToPrint.getCourriers().get(i);
+					Map<String, String> additionalFields = courrier.getVariables().getAdditionalFields();
+
+					// Create Element for each Map entry and append to the Variables node
+					additionalFields.entrySet().stream().sorted((entryA, entryB) -> entryA.getKey().compareToIgnoreCase(entryB.getKey()))
+							.forEach(
+									entry -> {
+										// Check if the element already exists
+										NodeList existingNodes = variablesElement.getElementsByTagName(entry.getKey());
+										if (existingNodes.getLength() > 0) {
+											// If the node exists, update its text content
+											existingNodes.item(0).setTextContent(entry.getValue());
+										} else {
+											// Otherwise, create a new element and append it
+											Element additionalElement = document.createElement(entry.getKey());
+											additionalElement.setTextContent(entry.getValue());
+											variablesElement.appendChild(additionalElement);
+										}
+									}
+							);
+				}
+			}
+
+			// Step 5: Write the Courriers XML file with expected name
+			String fileName = String.format("xml_d_%s_%s.xml", courriersToPrint.getCommunicationModel(), courriersToPrint.getEditionId());
+			Path outDirComm = Paths.get(outputDir + "/communication");
+			Path tempFilePath = outDirComm.resolve(fileName);
+
+			Transformer transformer = TransformerFactory.newInstance().newTransformer();
+			transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+			transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "2");
+			transformer.transform(new DOMSource(document), new StreamResult(Files.newOutputStream(tempFilePath)));
+
+			return tempFilePath;
+
+		} catch (JAXBException | IOException | ParserConfigurationException |
+				 javax.xml.transform.TransformerException e) {
+			e.printStackTrace();
+			return null;
```

the exception is swallowed here, need to do something about this